crates/rpc/src/server/api.rs

```diff
@@ -229,7 +232,11 @@ impl api_server::Api for RpcService {
         let tx_verifier = TransactionVerifier::new(MIN_PROOF_SECURITY_LEVEL);
 
         tx_verifier.verify(&tx).map_err(|err| {
-            Status::invalid_argument(format!("Invalid proof for transaction {}: {err}", tx.id()))
+            Status::invalid_argument(format!(
+                "Invalid proof for transaction {}: {}",
+                tx.id(),
+                err.as_report()
+            ))
```

Comment:
Would need to make `as_report_context()` a macro to support arbitrary format like this. Simpler to leave like this and use `as_report()`.